cmd/omniWatcher.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/geth"
+	"github.com/vulcanize/vulcanizedb/pkg/geth/client"
+	vRpc "github.com/vulcanize/vulcanizedb/pkg/geth/converters/rpc"
+	"github.com/vulcanize/vulcanizedb/pkg/geth/node"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/transformer"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/types"
+)
+
+// omniWatcherCmd represents the omniWatcher command
+var omniWatcherCmd = &cobra.Command{
+	Use:   "omniWatcher",
+	Short: "Watches events at the provided contract address",
+	Long: `Uses input contract address and event filters to watch events
+
+Expects an ethereum node to be running
+Expects an archival node synced into vulcanizeDB
+Requires a .toml config file:
+
+  [database]
+  name = "vulcanize_public"
+  hostname = "localhost"
+  port = 5432
+
+  [client]
+  ipcPath = "/Users/user/Library/Ethereum/geth.ipc"
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		omniWatcher()
+	},
+}
+
+func omniWatcher() {
+	if contractAddress == "" && len(contractAddresses) == 0 {
+		log.Fatal("Contract address required")
+	}
+
+	if !methodsOn && !eventsOn {
+		log.Fatal("Method polling and event watching turned off- nothing to do!")
+	}
+
+	if len(contractEvents) == 0 || len(contractMethods) == 0 {
+		var str string
+		for str != "y" {
+			reader := bufio.NewReader(os.Stdin)
+			if len(contractEvents) == 0 && len(contractMethods) == 0 {
+				fmt.Print("Warning: no events or methods specified, proceed to watch every event and method? (Y/n)\n> ")
+			} else if len(contractEvents) == 0 {
+				fmt.Print("Warning: no events specified, proceed to watch every event? (Y/n)\n> ")
+			} else {
+				fmt.Print("Warning: no methods specified, proceed to watch every method? (Y/n)\n> ")
+			}
+			resp, err := reader.ReadBytes('\n')
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			str = strings.ToLower(string(resp))
+			if str == "n" {
+				return
+			}
+		}
+	}
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	rawRpcClient, err := rpc.Dial(ipc)
+	if err != nil {
+		log.Fatal(fmt.Sprintf("Failed to initialize rpc client\r\nerr: %v\r\n", err))
+	}
+
+	rpcClient := client.NewRpcClient(rawRpcClient, ipc)
+	ethClient := ethclient.NewClient(rawRpcClient)
+	client := client.NewEthClient(ethClient)
+	node := node.MakeNode(rpcClient)
+	transactionConverter := vRpc.NewRpcTransactionConverter(ethClient)
+	blockChain := geth.NewBlockChain(client, node, transactionConverter)
+	db, err := postgres.NewDB(databaseConfig, blockChain.Node())
+	if err != nil {
+		log.Fatal(fmt.Sprintf("Failed to initialize database\r\nerr: %v\r\n", err))
+	}
+
+	con := types.Config{
+		DB:      db,
+		BC:      blockChain,
+		Network: network,
+	}
+
+	t := transformer.NewTransformer(&con)
+
+	contractAddresses = append(contractAddresses, contractAddress)
+	for _, addr := range contractAddresses {
+		t.SetEvents(addr, contractEvents)
+		t.SetMethods(addr, contractMethods)
```

Comment:
Do we want to specify and validate event and method names that are acceptable parameters for this command? Thinking things could get hairy - with the approach of appending all events and methods to every address - if folks specified multiple contract addresses with events or methods that were unique to individual contracts.

Separate note, is there a board where stories are being maintained for this work? Wondering about where to document potential future stories (e.g. specifying different events/methods for contracts while simultaneously watching multiple contracts)

Reply:
Yeah we will definitely want to be able to configure event/methods sets for each contract. In the meantime it might be best to leave out the ability to watch multiple addresses?

No board yet afaik but that would very nice to have.